src/main/java/org/openmaptiles/util/Utils.java

```diff
@@ -12,10 +13,25 @@ public static <T> T coalesce(T a, T b) {
     return a != null ? a : b;
   }
 
+  public static <T, U> T coalesceF(T a, Function<U, T> fb, U b) {
+    return a != null ? a : fb.apply(b);
+  }
+
   public static <T> T coalesce(T a, T b, T c) {
     return a != null ? a : b != null ? b : c;
   }
 
+  public static <T, U> T coalesceF(T a, Function<U, T> fb, U b, Function<U, T> fc, U c) {
```

Comment:
`coalesceF` is a little cryptic, although it makes sense more sense when you see the signature.  Would `coalesceLazy` make sense?

Reply:
Renamed.